Sources/SignalProducer.swift

```diff
@@ -1877,6 +1877,59 @@ extension SignalProducerProtocol {
 			}
 		}
 	}
+	
+	/// Creates a new `SignalProducer` that will bucket each received value into
+	/// a group based on the key returned from `grouping`. 
+	///
+	/// Termination events on the original signal are also forwarded to each
+	/// producer group.
+	///
+	/// - parameters:
+	///   - grouping: a closure that determines the grouping key for a given value
+	///
+	/// - returns: A producer of producers amits one producer for each group and forwards
+	///            each value from the original producer to the inner producer corresponding 
+	///            to the group to which the value belongs to (as determined by the key)
+	public func groupBy<Key: Hashable>(_ grouping: @escaping (Value) -> Key) -> SignalProducer<(Key, SignalProducer<Value, Error>), Error> {
```

Comment:
I've tried that for a short moment, but on the call site that would then look like `producer.group { ... }` instead of `producer.groupBy { ... } `when using trailing closure syntax. 
Not sure if thats a problem though...